tests/benchmarks/micro_benchmarks/test_cuda_nccl_bw_performance.py

```diff
@@ -0,0 +1,353 @@
+
+"""Tests for nccl-bw benchmark."""
+
+import os
+import numbers
+import unittest
+from pathlib import Path
+
+from superbench.benchmarks import BenchmarkRegistry, BenchmarkType, ReturnCode, Platform
+
+
+class CudaNcclBwBenchmarkTest(unittest.TestCase):
+    """Tests for CudaNcclBwBenchmark benchmark."""
+    def setUp(self):
+        """Method called to prepare the test fixture."""
+        # Create fake binary file just for testing.
+        os.environ['SB_MICRO_PATH'] = '/tmp/superbench/'
+        binary_path = os.path.join(os.getenv('SB_MICRO_PATH'), 'bin')
+        Path(binary_path).mkdir(parents=True, exist_ok=True)
+        self.__binary_files = []
+        for bin_name in ['all_reduce_perf', 'all_gather_perf', 'broadcast_perf', 'reduce_perf', 'reduce_scatter_perf']:
+            self.__binary_files.append(Path(binary_path, bin_name))
+            Path(binary_path, bin_name).touch(mode=0o755, exist_ok=True)
+
+    def tearDown(self):
+        """Method called after the test method has been called and the result recorded."""
+        for binary_file in self.__binary_files:
+            binary_file.unlink()
+
+    def test_nccl_bw_performance(self):
+        """Test nccl-bw benchmark."""
+        benchmark_name = 'nccl-bw'
+        (benchmark_class,
+         predefine_params) = BenchmarkRegistry._BenchmarkRegistry__select_benchmark(benchmark_name, Platform.CUDA)
+        assert (benchmark_class)
+
+        benchmark = benchmark_class(benchmark_name)
+
+        ret = benchmark._preprocess()
+        assert (ret is True)
+        assert (benchmark.return_code == ReturnCode.SUCCESS)
+
+        # Check basic information.
+        assert (benchmark)
+        assert (benchmark.name == 'nccl-bw')
+        assert (benchmark.type == BenchmarkType.MICRO)
+
+        # Check parameters specified in BenchmarkContext.
+        assert (benchmark._args.algo == ['allreduce', 'allgather', 'broadcast', 'reduce', 'reducescatter'])
+        assert (benchmark._args.gpu_count == 8)
+
+        # Check command list
+        bin_names = ['all_reduce_perf', 'all_gather_perf', 'broadcast_perf', 'reduce_perf', 'reduce_scatter_perf']
+        print(benchmark._commands)
+        for i in range(len(benchmark._args.algo)):
+            commnad = bin_names[i] + benchmark._commands[i].split(bin_names[i])[1]
+            expected_command = '{} -b 1 -e {} -f 2 -g {} -c 0'.format(
+                bin_names[i], benchmark._args.max_size, benchmark._args.gpu_count
+            )
+            assert (commnad == expected_command)
+
+        # Check results and metrics.
+        # Case with no raw_output
+        assert (benchmark._process_raw_result(0, '') is False)
+
+        # Case with valid raw_output
+        raw_output = {}
+        raw_output['allgather'] = """
+# nThread 1 nGpus 8 minBytes 1 maxBytes 8589934592 step: 2(factor) warmup iters: 5 iters: 20 validation: 0
+#
+# Using devices
+#   Rank  0 Pid 112372 on sb-validation-scus-000001 device  0 [0x00] A100-SXM4-40GB
```

Comment:
remove all host related strings? you can mock it to `localhost`

Reply:
fixed